packages/events/src/util.ts

```diff
@@ -55,3 +56,48 @@ export function getEventServiceUrl(
     }
   });
 }
+
+/**
+ * Get the events query based on type.
+ * @param type - string to indicate event type with the options `upcoming`, `past`, `cancelled` and `draft`
+ * @param params - query features request options
+ * @param requestOptions - request options that may include authentication
+ * @returns an IQueryFeaturesRequestOptions with modified Where and OrderBy params
+ */
+export function getEventQueryFromType(
+  type: string,
+  params: IQueryFeaturesRequestOptions,
+  requestOptions?: IRequestOptions
+): IQueryFeaturesRequestOptions {
```

Comment:
when a method has more than one _required_ input parameter, the (undocumented) convention we've agreed on is to stuff them all into a custom `IRequestOptions` interface.
```ts
export interface IGetEventQueryRequestOptions extends IQueryFeaturesRequestOptions {
  eventType: string;
}

export function getEventQueryFromType(requestOptions: IGetEventQueryRequestOptions): any
```

Reply:
This is a helper util though, so we don't need to follow that convention here.

The signature as defined is misleading. I'm not sure this should return a `IQueryFeaturesRequestOptions`. Seems to me it should return just `{ where: string; orderByFields: string; }`

Reply:
How about if there is nothing returned at all? The function actually manipulates the input `IQueryFeaturesRequestOptions`

Reply:
The function should not have side effects if avoidable. Whatever it needs to change should be returned. Then the caller can do

```js
newOptions = getEventQueryFromType(type, originalOptions);
queryFeatures({...originalOptions, ...newOptions});
```